ensenso_camera/src/virtual_object_handler.cpp

```diff
@@ -41,6 +153,18 @@ VirtualObjectHandler::VirtualObjectHandler(const std::string &filename, const st
     for (int i = 0; i < objects.count(); ++i) {
         originalPoses.push_back(poseFromNxLib(objects[i][itmLink]));
     }
+
+    // Create publisher thread
+    if (!markerTopic.empty()) {
+      markerThread = std::thread([=](){
+        VirtualObjectMarkerPublisher publisher{markerTopic, markerPublishRate, objects, objectsFrame, stopMarkerThread};
+        publisher.run();
+      });
+    }
+}
+
+VirtualObjectHandler::~VirtualObjectHandler() {
+  stopMarkerThread = true;
```

Comment:
You will have to join the _markerThread_ if the thread is joinable. Otherwise [std::terminate](https://en.cppreference.com/w/cpp/error/terminate) (point 10) will be called.
```cpp
if (markerThread.joinable()){
    markerThread.join();
}
```

Reply:
Agreed, good catch